### PR TITLE
[WIP] Implement dispatchInOrder

### PR DESCRIPTION
**Pull request by erlingfn**
Create a utility function for dispatching actions one after the other, i. e. dispatching fetchEvent before isUserFollowing

### CODE REVIEW

File: app/routes/events/EventDetailRoute.js [hunk 1]
```diff
@@ -124,11 +125,14 @@ const mapDispatchToProps = {
...
 };
-
+/*
 const loadData = async ({ params: { eventId }, loggedIn }, dispatch) => [
   await dispatch(fetchEvent(eventId)),
   loggedIn && (await dispatch(isUserFollowing(eventId)))
 ];
+*/
+const loadData = async ({ params: { eventId }, loggedIn }, dispatch) =>
+  await dispatchInOrder([fetchEvent, isUserFollowing], dispatch, eventId);
```
Comment (orhanhenrik) [inline app/routes/events/EventDetailRoute.js]: In this refactoring you are missing the `loggedIn` check before calling `isUserFollowing`. Is that intentional?

File: app/routes/pages/PageDetailRoute.js [hunk 2]
```diff
@@ -109,11 +110,8 @@ const loadData = async (props, dispatch) => {
         .concat(dispatch(fetchAllPages()))
     );
   }
-  const itemActions = [];
+  const itemActions = dispatchInOrder(fetchItemActions, dispatch, pageSlug);
```
Comment (orhanhenrik) [inline app/routes/pages/PageDetailRoute.js]: In this case, do you really need to dispatch them in exact order? Can't all `fetchItem`s be called at the same time?

File: app/routes/events/EventDetailRoute.js (hunk 1 above)
Comment (erlingfn) [reply app/routes/events/EventDetailRoute.js]: It is not, nice catch.

File: app/routes/pages/PageDetailRoute.js (hunk 2 above)
Comment (erlingfn) [reply app/routes/pages/PageDetailRoute.js]: I believe the committee actions have to be dispatched consecutively, as you need a group in the state in order to mutate it's membership.

**Comment by erlingfn** [comment]
I agree that forcing all actions to take the same argument restricts the cases where this abstraction would be useful. The original thought behind it was that I saw two similar errors with very similar solutions, and that an abstraction would be necessary if we encountered more in order to not have to tailor make a solution every time. Does anyone have any thoughts on whether or not we need a general solution for this?

File: app/routes/pages/PageDetailRoute.js (hunk 2 above)
Comment (erlingfn) [reply app/routes/pages/PageDetailRoute.js]: I believe the committee actions have to be dispatched consecutively, as you need a group in the state in order to mutate it's membership.

Comment (erlingfn) [reply app/routes/pages/PageDetailRoute.js]: I believe the committee actions have to be dispatched consecutively, as you need a group in the state in order to mutate it's membership.

**Comment by mrfarstad** [comment]
Cool suggestion! This will definitely make the code shorter, and I can't say that I vote against it 🌮

**Comment by mrfarstad** [comment]
Have this PR been discussed further? Do we think this abstraction is necessary? ☯️

**Comment by erlingfn** [comment]
I have not found any more places to use this, and since it is not very flexible I recommend that we scrap this PR. If a utility function becomes necessary in the future it is not difficult to implement

**Pull request closed without merging**